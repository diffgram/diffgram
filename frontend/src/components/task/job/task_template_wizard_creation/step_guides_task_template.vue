<template>
  <v-container fluid data-cy="task-template-guide-step">
    <div class="d-flex mb-8 justify-space-between">
      <h1 data-cy="task-template-guide-step-title" class="font-weight-medium text--primary mr-4">
        Guides Setup:
      </h1>
      <standard_button
        tooltip_message="Create Guides"
        @click="open_guides"
        button_color="primary"
        icon="mdi-plus"
        button_message="Create Guides"
        color="white">
      </standard_button>
    </div>

    <v_error_multiple :error="error"></v_error_multiple>
    <p class="text--primary">
      Guides are used to explain annotators the requirements for correctly annotating your files.
      This is an optional step.
    </p>


    <v-container fluid>
      <h2 class="ma-0" data-cy="task-template-guide-step-subtitle">Set Your Guide (Optional): </h2>
      <guide_selector
        data-cy="guide-selector"
        :project_string_id="project_string_id"
        @change="on_change_guide"
        ref="guide_selector"
      >

      </guide_selector>
    </v-container>

    <wizard_navigation
      @next="on_next_button_click"
      :disabled_next="loading"
      :loading_next="loading_steps"
      @back="$emit('previous_step')"
      :skip_visible="false"
    >
    </wizard_navigation>

    <v-dialog v-model="dialog_open"
              :close-on-content-click="false"
              :nudge-width="200"
              offset-x>


      <v_guide_new_or_edit
        :project_string_id="project_string_id"
        :mode="'new'"
        @guide_new_success="on_guide_created">
      </v_guide_new_or_edit>

    </v-dialog>
  </v-container>

</template>

<script lang="ts">

  import axios from '../../../../services/customInstance';
  import guide_selector from '../../guide/guide_selector'

  import Vue from "vue";

  export default Vue.extend({
      name: 'step_guides_task_template',
      props: [
        'project_string_id',
        'job',
        'loading_steps'
      ],

      components: {
        guide_selector
      },

      data() {
        return {
          error: {},
          show_credentials: false,
          dialog_open: false,
          loading: false,
        }
      },
      created() {

      },

      methods: {
        on_change_guide: function (guide) {
          this.job.guide = guide;
          this.attach_selected(guide.id, 'default', 'update')
        },
        on_next_button_click: function () {
          this.$emit('next_step');
        },
        open_guides: function () {
          this.dialog_open = true;
        },
        on_guide_created: function (guide) {
          this.$refs.guide_selector.guide_list_api();
        },
        attach_selected: async function (guide_id, attach_kind, update_or_remove) {


          this.loading = true
          this.show_success_attach = false
          this.error = {}
          try {
            const response = await axios.post('/api/v1/guide/attach/job',
              {
                'update_or_remove': update_or_remove,
                'kind': attach_kind,
                'job_id': this.job.id,
                'guide_id': guide_id

              })

          } catch (e) {
            this.error = e.response.data.log.error
            console.error(e)
          }
          finally {
            this.loading = false;
          }

        },
      }
    }
  ) </script>
