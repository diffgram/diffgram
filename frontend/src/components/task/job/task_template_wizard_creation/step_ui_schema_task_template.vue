<template>
  <v-container fluid data-cy="task-template-ui-schema-step">
    <div class="d-flex mb-8 justify-space-between">
      <h1 data-cy="task-template-ui-schema-step-title" class="font-weight-medium text--primary mr-4">
        White-label Customization (BETA)
      </h1>
      <standard_button
        tooltip_message="Launch UI Editor"
        @click="open_ui_schema_creation"
        button_color="primary"
        icon="mdi-puzzle-edit"
        :left="true"
        button_message="Launch UI Editor"
        color="white">
      </standard_button>
    </div>

    <v_error_multiple :error="error"></v_error_multiple>
    <p class="text--primary">
      UI Schemas are a way to Customize the Annotation UI. You can show/hide buttons and configure
      what you want the annotators to see beyond Label Schema.
    </p>

    <v-container fluid>
      <ui_schema_selector
        data-cy="ui-schema-selector"
        :project_string_id="project_string_id"
        :show_default_option="true"
        @change="on_change_schema">

      </ui_schema_selector>

    </v-container>

    <wizard_navigation
      @next="on_next_button_click"
      :loading_next="loading_steps"
      :disabled_next="loading_steps"
      @back="$emit('previous_step')"
      :skip_visible="false"
    >
    </wizard_navigation>
  </v-container>

</template>

<script lang="ts">

import ui_schema_selector from '../../../ui_schema/ui_schema_selector'

import Vue from "vue";

export default Vue.extend({
  name: 'step_ui_schema_task_template',
  props: [
    'project_string_id',
    'job',
    'loading_steps',
  ],

  components: {
    ui_schema_selector,
  },

  data() {
    return {
      error: {},
      open_wizard: false,
    }
  },
  created() {

  },

  computed: {},
  methods: {
    on_change_schema: function (ui_schema) {
      this.job.ui_schema_id = ui_schema.id;
    },
    on_next_button_click: function () {
      this.$emit('next_step');
    },
    open_ui_schema_creation: function () {
      let routeData = this.$router.resolve({
        path:  `/task/-1`,
        query: {
          edit_schema: true,
          create_new_on_load: true,
          view_only: true
          }
      });
      window.open(routeData.href, '_blank');
    },
  }
}
  ) </script>
