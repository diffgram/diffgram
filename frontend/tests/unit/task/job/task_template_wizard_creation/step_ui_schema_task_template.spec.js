import {shallowMount, createLocalVue} from "@vue/test-utils";
import Vuex from 'vuex'
import axios from 'axios'
import step_ui_schema_task_template
  from "../../../../../src/components/task/job/task_template_wizard_creation/step_ui_schema_task_template";
import VueRouter from 'vue-router'
import step_guides_task_template
  from "../../../../../src/components/task/job/task_template_wizard_creation/step_guides_task_template";

const localVue = createLocalVue();
localVue.use(Vuex)
localVue.use(VueRouter)
const router = new VueRouter()

let job = {
  name: 'test',
  label_mode: 'closed_all_available',

  loading: false,
  passes_per_file: 1,
  share_object: {
    // TODO this may fail for org jobs? double check this.
    'text': String,
    'type': 'project'
  },
  share: 'project',
  instance_type: 'box', //"box" or "polygon" or... "text"...
  permission: 'all_secure_users',
  field: 'Other',
  category: 'visual',
  attached_directories_dict: {attached_directories_list: []},
  type: 'Normal',
  connector_data: {},
  // default to no review while improving review system
  review_by_human_freqeuncy: 'No review', //'every_3rd_pass'
  td_api_trainer_basic_training: false,
  file_handling: "use_existing",
  interface_connection: undefined,
  member_list_ids: ["all"]
}
describe("step_ui_schema_task_template.vue", () => {
  let actions;
  let store;

  beforeEach(() => {
    actions = {
      actionClick: jest.fn(),
      actionInput: jest.fn()
    }
    store = new Vuex.Store({
      actions
    })
  });


  it("Renders component correctly.", () => {
    const wrapper = shallowMount(step_ui_schema_task_template, {
      store,
      localVue,
      propsData: {
        job: job
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("changes schema when calling  on_change_schema()", () => {
    const wrapper = shallowMount(step_ui_schema_task_template, {
      store,
      localVue,
      propsData: {
        job: job
      }
    });
    wrapper.vm.on_change_schema({id: 'test'})
    expect(wrapper.vm.$props.job.ui_schema_id).toEqual('test')

  });

  it("Emits event when calling on_next_button_click()", () => {
    const wrapper = shallowMount(step_ui_schema_task_template, {
      store,
      localVue,
      propsData: {
        job: job
      }
    });
    wrapper.vm.on_next_button_click()
    expect(wrapper.emitted().next_step.length).toBe(1);
  });

  it("Resolver route when calling open_ui_schema_creation()", () => {
    const wrapper = shallowMount(step_ui_schema_task_template, {
      store,
      localVue,
      router,
      propsData: {
        job: job
      }
    });
    const spy = jest.spyOn(wrapper.vm.$router, 'resolve')
    wrapper.vm.open_ui_schema_creation()
    expect(spy).toHaveBeenCalled()
  });





});
