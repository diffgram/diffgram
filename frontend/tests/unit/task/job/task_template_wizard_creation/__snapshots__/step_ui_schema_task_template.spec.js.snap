// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`step_ui_schema_task_template.vue Renders component correctly. 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
